_specifications/specification-3-17.md

```diff
@@ -8915,11 +8930,125 @@ Servers usually support different communication channels (e.g. stdio, pipes, ...
+
+> *Since version 3.17.0*
+
+The request is sent from the client to the server to compute inlay hints for a given text document. These hints can provide interstitial information about the code that is not part of the language syntax, like names of arguments, inferred type information, etc.
+
+_Client Capability_:
+* property name (optional): `textDocument.inlayHints`
+* property type: `InlayHintsClientCapabilities` defined as follows:
+
+<div class="anchorHolder"><a href="#inlayHintsClientCapabilities" name="inlayHintsClientCapabilities" class="linkableAnchor"></a></div>
+
+```typescript
+export interface InlayHintsClientCapabilities {
+	/**
+	 * Whether inlay hints supports dynamic registration.
+	 */
+	dynamicRegistration?: boolean;
+}
+```
+
+_Server Capability_:
+* property name (optional): `inlayHintsProvider`
+* property type: `boolean | InlayHintsProvider | InlayHintsRegistrationOptions` is defined as follows:
+
+<div class="anchorHolder"><a href="#inlayHintsOptions" name="inlayHintsOptions" class="linkableAnchor"></a></div>
+
+```typescript
+export interface InlayHintsOptions extends WorkDoneProgressOptions {
+}
+```
+
+_Registration Options_: `InlayHintsRegistrationOptions` defined as follows:
+
+<div class="anchorHolder"><a href="#inlayHintsRegistrationOptions" name="inlayHintsRegistrationOptions" class="linkableAnchor"></a></div>
+
+```typescript
+export interface InlayHintsRegistrationOptions extends TextDocumentRegistrationOptions, InlayHintsOptions {
+}
+```
+
+_Request_:
+
+* method: `textDocument/inlayHints`
+* params: `InlayHintsParams` defined as follows:
+
+<div class="anchorHolder"><a href="#inlayHintsParams" name="inlayHintsParams" class="linkableAnchor"></a></div>
+
+```typescript
+export interface InlayHintsParams extends WorkDoneProgressParams, PartialResultParams {
+	/**
+	 * The document to request inlay hints for.
+	 */
+	textDocument: TextDocumentIdentifier;
+
+	/**
+	 * The range to request inlay hints for.
+	 */
+	range: Range;
+}
+```
+
+<div class="anchorHolder"><a href="#inlayHintKind" name="inlayHintKind" class="linkableAnchor"></a></div>
+
+```typescript
+export declare namespace InlayHintKind {
+	export const Other = 0;
+	export const Type = 1;
+	export const Parameter = 2;
+}
+
+export type InlayHintKind = 0 | 1 | 2;
+```
+
+_Response_:
+* result: `InlayHint[]` \| `null` defined as follows:
+
+<div class="anchorHolder"><a href="#inlayHint" name="inlayHint" class="linkableAnchor"></a></div>
+
+```typescript
+/**
+ * Inlay hint information.
+ */
+export interface InlayHint {
+	/**
+	 * The text of the hint.
+	 */
+	text: string;
+
+	/**
+	 * The position of this hint.
+	 */
+	position: Position;
+
+	/**
+	 * The kind of this hint.
+	 */
+	kind?: InlayHintKind;
+
+	/**
+	 * Whitespace before the hint.
+	 */
+	whitespaceBefore?: boolean;
+
+	/**
+	 * Whitespace after the hint.
+	 */
+	whitespaceAfter?: boolean;
```

Comment:
Can we include a note in the comment that states what happens when these optional fields are omitted?